kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/CachingKms.java

```diff
@@ -33,6 +40,9 @@ public class CachingKms<K, E> implements Kms<K, E> {
     private final Kms<K, E> delegate;
     private final AsyncLoadingCache<E, SecretKey> decryptDekCache;
     private final AsyncLoadingCache<String, K> resolveAliasCache;
+    private final Cache<String, Long> notFoundAliasCache;
+    private static final Duration CACHE_NOT_FOUND_ALIAS_DURATION = Duration.ofSeconds(30);
```

Comment:
I think we should make this configurable, and support turning it off entirely.

Reply:
I think we should leave it hard coded for now as it might well go away once we fix the config model. Is it really something we want to surface in the filter's public API as is?

If we do keep this or variation of it once the we fix the model then yes we should definitely allow users to tune this.

Reply:
> it might well go away once we fix the config model

Will it though? It seems like this is a useful thing to prevent someone/somethingfrom putting unnecessary load on the KMS. In particular, once `UnknownAliasException` is no longer being treated as "no encryption", but instead propagates to the client as an error to indicate "sort your config/provision your KEK" and the client is coded to retry that error in a tight loop.

Reply:
For other exceptions/response codes we would go into our exponential backoff code and do a couple of retries with delay before giving up, throwing exception and closing the connection.

I see we could cache arbitrary failures to prevent a storm of reconnecting clients from hammering the KMS. Though maybe it's better addressed by some kind of overall rate limit or circuitbreaker around the KMS, #794 might be the place to discuss.

Reply:
I've exposed the cache config to users (described in the issue description). The existing config class looked like the right place to put it so I've exposed all the properties.